### ISSUE TITLE
Playback with no audio output

### ISSUE THREAD

**Issue by MiltonKeynes** [ilia3101/MLV-App#192]
From some time I have been unable to playback MLV's with sound. MLV App actually shows the files with the green audio track and data on it. When exporting in any of the formats and opening the file in VLC, then I can hear the audio with the video. So, again, the issue happens only when trying to playback my files in MLV App. It happened to me with both 1.8 and 1.9 versions. Here are my specs:

Windows 10 Pro (compilation 18363)
IntelCore i7-6700 3.40 GHz
16,0 GB RAM

**Comment by masc4ii** [comment]
The problem is known. But until now nobody was able to evaluate if it is a Windows, Qt or MLVApp problem. I am not able to evaluate this, because here it works fine on all computers and platforms I have around me.

**Comment by bouncyball-git** [comment]
Try to use static windows build. Someone on the forum reported to have the probs you have and this fixed his issues.

**Comment by MiltonKeynes** [comment]
> Try to use static windows build (there is older QT). Someone on the forum reported to have the probs you have and this fixed his issues.

It worked for me. Thanks so much.

**Comment by masc4ii** [comment]
Oh really? Hm... here it works with all versions. I am searching in the dark...

**Comment by bouncyball-git** [comment]
Yeah... I know. Only a few people complained about no audio after we released 1.8. This is Mystery for me too.

**Comment by JGSphaela** [comment]
Also stumbled upon this issue when I tried MLV App v1.14(Qt version 5.15.2) on Arch Linux(Linux 6.5.5-lqx1-1-lqx). No audio what so ever. MLV App was built via yay, and I use pipewire for audio. Everything else works just fine.

**Comment by esbante** [comment]
This happened to me just now today. No audio in the latest appimage, or in the version built from source. I'm using Ubuntu 22.04. Earlier today everything was working fine.

**Comment by masc4ii** [comment]
@esbante : are you in "drop frame mode" and activated "audio output" from menu? Otherwise there is no audio playback.

**Comment by esbante** [comment]
![Skärmbild från 2023-10-11 12-25-18](https://github.com/ilia3101/MLV-App/assets/34633619/87e06e2b-8461-404e-8fb1-31e1ad4b14c7)

Yes, but should the audio output icon look like this (haven't looked at it before)? Clicking it gives a little square around it, which makes no difference, but it is hard to tell if it is on or not

**Comment by esbante** [comment]
![Skärmbild från 2023-10-11 12-27-55](https://github.com/ilia3101/MLV-App/assets/34633619/751cbfd0-d57d-4bf4-99e1-0244d3035592)

Like so

**Comment by esbante** [comment]
I have used the mlv-app for several years, but never come across this. It happened while I was switching between my USB-C dock, so the system audio was changing betweeen that and my laptop, if that could have messed something up? Same behaviour in the appimage as in the built version.

**Comment by esbante** [comment]
Ok, the sound in the appimage is working as it should now, so it is just the built version, which makes more sense. Maybe I should just rebuild it and see if it starts working.

**Comment by JGSphaela** [comment]
I dig into this problem and found something that might be related or not:

1. My system has Anaconda installed. If compiled using conda-forge gcc(13.2.0) it will not have audio playback. If compiled use /usr/bin/gcc(13.2.1) it works fine.
2. running MLV-App under sudo will not have audio playback. Both on Appimage and built version. But on built version it seems like it's require to run under sudo to install focus pixel maps, so it's needed to running sudo first to install the maps then used it to have audio playback.

I'm on Arch Linux 6.5.10-lqx1-1-lqx with pipeaudio. I also discovered that focus pixel maps for some reason, not working on extracted Appimage version although it shows installed. But it's not related here so I might open another issue.

**Comment by nwroot** [comment]
Found a solution!

If you open mlvapp on the terminal, upon trying to playback with audio, you get the following message

`using null output device, none available`

A quick google reveals this is usually caused by missing plugins, which are included on the AppImage, but may not be installed for manually compiled builds.

The solution? On debian based distros, install `libqt5multimedia5-plugins`, which includes the required plugins

**Comment by masc4ii** [comment]
@nwroot Thank you for the hint. I think this plugin should be included in the appimage, to not get Qt and library version mismatches. I can install it in the github runner, which builds the appimage. But I expect it not to be included in the appimage, if it is not explicitely used by the app, so not automatically packed by Qt. Any idea how to force include it using the .pro project file?

**Comment by masc4ii** [comment]
@nwroot Could you please test the latest Linux runner output, without the plugin package installed? I tried to explicitely include it, but can't test.
https://github.com/ilia3101/MLV-App/actions/runs/12613218556

**Comment by nwroot** [comment]
> @nwroot Could you please test the latest Linux runner output, without the plugin package installed? I tried to explicitely include it, but can't test. https://github.com/ilia3101/MLV-App/actions/runs/12613218556

Tested with the package uninstalled and the audio works and the plugins are indeed present.

```
$ ls /tmp/.mount_MLVAppXFrIFd/usr/plugins/*
/tmp/.mount_MLVAppXFrIFd/usr/plugins/audio:
libqtaudio_alsa.so  libqtmedia_pulse.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/bearer:
libqconnmanbearer.so  libqgenericbearer.so  libqnmbearer.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/imageformats:
libqgif.so  libqico.so  libqjpeg.so  libqsvg.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/mediaservice:
libgstaudiodecoder.so  libgstcamerabin.so  libgstmediacapture.so  libgstmediaplayer.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/platforminputcontexts:
libcomposeplatforminputcontextplugin.so  libibusplatforminputcontextplugin.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/platforms:
libqxcb.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/playlistformats:
libqtmultimedia_m3u.so

/tmp/.mount_MLVAppXFrIFd/usr/plugins/xcbglintegrations:
libqxcb-egl-integration.so  libqxcb-glx-integration.so
```
Although for me the audio did work on the AppImage builds before, so someone that had the issue even with the AppImage should also try this build.

I originally had the issue with a manually built version, which got fixed by installing `libqt5multimedia5-plugins`, maybe it should be added to the build instructions?

**Comment by masc4ii** [comment]
Thank you. Done. 65273ecf0124c4dd29c1296767a654fc96ae2cd2

**Comment by esbante** [comment]
Great! I did not have the sound problem in the appimage version, just in the manual build, but with installation of libqt5multimedia5-plugins it works as it should. Thanks!
